gammapy/makers/background/fov.py

```diff
@@ -136,19 +136,32 @@ def _verify_requirements(self, dataset):
         """"Verify that the requirements of min_counts
         and min_npred_background are satisfied"""
 
+        npred = dataset.npred()
         mask = dataset.mask
+        mask &= ~np.isnan(npred)
         count_tot = dataset.counts.data[mask].sum()
+        npred_tot = npred.data[mask].sum()
         bkg_tot = dataset.npred_background().data[mask].sum()
+        not_bkg_tot = npred_tot - bkg_tot
 
-        if count_tot <= self.min_counts:
+        self._value_cached = (count_tot - not_bkg_tot) / bkg_tot
```

Comment:
Does the caching here really improve any performance? I'd be surprised if this really represents any bottleneck. Otherwise I'd simplify and remove...

To avoid the code duplication for the masked counts computation once could introduce `make_masked_summed_counts()`, which returns a dict like `{"npred_total", "counts_total", "background_total"}`

Reply:
done